### DISCUSSION TITLE
Always get the same output from LLM regardless of seed/random seed

### DISCUSSION THREAD

**Discussion by vitalifa** [oobabooga/text-generation-webui#2425] [Q&A] [answered]
I am using an instruction model (Guanaco 33B) with the "Guanaco non-chat" instruction format. I always get an identical output for a given prompt, no matter what the seed is set to. Even at seed = -1 (random), it will generate identical output as though a fixed seed is being used.

Am I missing something?

Is this a limitation of instruct-type models perhaps?

Please share if you have discovered any tricks around generating different outputs for the same instruction.

**Comment by matatonic** [comment]
You can try adjusting the temperature (higher, try 1.99 and work down), and/or other generating parameters such as top_p (try 0.1 and work up)  or top_k (try 100 and work down), a pretty good example of how they work is here: https://docs.cohere.com/docs/controlling-generation-with-top-k-top-p

**Reply by vitalifa** [reply, marked as answer]
This was immensely helpful!

As you suggested, I began getting differing output after adjusting the "top_p" to be higher. This one parameter made all the difference. TYSM :+1:
